### DISCUSSION TITLE
General handler for 500

### DISCUSSION THREAD

**Discussion by iamthen0ise** [fastapi/fastapi#4798] [Questions] [answered]
Hey!

I have a very simple question. Can I somehow wrap all exceptions so that for any 500 error, FastAPI can return a JSON response with a body like `{"code": 500, "msg": "Internal Server Error"}` ?

Thank you!

**Comment by accelleon** [comment, marked as answer]
You should be able to do this:
```python
from fastapi import FastAPI, Request
from fastapi.encoders import jsonable_encoder
from fastapi.responses import JSONResponse

app = FastAPI()

@app.exception_handler(500)
async def internal_exception_handler(request: Request, exc: Exception):
  return JSONResponse(status_code=500, content=jsonable_encoder({"code": 500, "msg": "Internal Server Error"}))
```

**Reply by iamthen0ise** [reply]
Thank you!

**Reply by chipun** [reply]
@accelleon The unit test did not work because it ran without triggering the global exception handler
